examples/rtspDeviceShifu/camera.py

```diff
@@ -25,75 +39,149 @@
     "right": CAMERA_CTRL_MOVE_RIGHT,
 }
 
-
 class VideoGet:
-    """
-    Class that continuously gets frames from a VideoCapture object
-    with a dedicated thread.
-    """
-
     def __init__(self, ip, rtsp_port, username, password):
-        self.stream = cv2.VideoCapture("rtsp://{}:{}@{}{}".format(CAMERA_USERNAME, CAMERA_PASSWORD, ip, rtsp_port))
-        (self.grabbed, self.frame) = self.stream.read()
-        self.stopped = False
+        self.ip = ip
+        self.rtsp_port = rtsp_port
+        self.username = username
+        self.password = password
+        self.stream = None
+        self.frame_queue = queue.Queue(maxsize=BUFFER_SIZE)
+        self._should_stop = False
+        self._capture_thread = None
+        self._last_frame = None
+        self._client_count = 0
+        self._last_gc_time = time.time()
+        self._reconnect_attempts = 0
+        self._max_reconnect_attempts = 5
+        self._reconnect_delay = 1  # seconds
+        self.start()
+
+    def _reconnect(self):
+        if self.stream is not None:
+            self.stream.release()
+        
+        rtsp_url = f"rtsp://{self.username}:{self.password}@{self.ip}{self.rtsp_port}"
+        logger.info(f"Attempting to reconnect to RTSP stream: {rtsp_url}")
+        
+        self.stream = cv2.VideoCapture(rtsp_url)
+        if not self.stream.isOpened():
+            logger.error("Failed to reconnect to RTSP stream")
+            return False
+        logger.info("Successfully reconnected to RTSP stream")
+        return True
+
+    def _capture_frames(self):
+        frame_count = 0
+        last_fps_time = time.time()
+        
+        while not self._should_stop:
+            try:
+                if self._client_count == 0:
+                    time.sleep(0.1)
+                    continue
+
+                if self.stream is None or not self.stream.isOpened():
+                    if self._reconnect_attempts < self._max_reconnect_attempts:
+                        if self._reconnect():
+                            self._reconnect_attempts = 0
+                        else:
+                            self._reconnect_attempts += 1
+                            time.sleep(self._reconnect_delay)
+                            continue
+                    else:
+                        logger.error("Max reconnection attempts reached")
+                        time.sleep(1)
+                        continue
+
+                success, frame = self.stream.read()
+                if not success:
+                    logger.warning("Failed to read frame, attempting reconnection")
+                    if not self._reconnect():
+                        time.sleep(self._reconnect_delay)
+                        continue
+                    continue
+
+                if frame.shape[1] != CAMERA_WIDTH or frame.shape[0] != CAMERA_HEIGHT:
```

Comment:
consider using a resize ratio for reducing the frame size instead of fixed width and height